types/framework/entities/actor.d.ts

```diff
@@ -294,47 +294,56 @@ declare class Actor<
   /** @override */
   createEmbeddedEntity(
     embeddedName: string,
-    data: Optional<ActiveEffect | Actor.OwnedItem<I>>,
+    data: Optional<ActiveEffect.Data | Actor.OwnedItem<I>>,
     options?: any
-  ): Promise<ActiveEffect | Actor.OwnedItem<I>>;
+  ): Promise<ActiveEffect.Data | Actor.OwnedItem<I>>;
 
   /**
    * When Owned Items are created process each item and extract Active Effects to transfer to the Actor.
    * @param created - Created owned Item data objects
    * @param temporary - Is this a temporary item creation?
    * @returns An array of effects to transfer to the Actor
    */
-  _createItemActiveEffects(created: ActiveEffect, { temporary }?: { temporary?: boolean }): ActiveEffect;
+  _createItemActiveEffects(
+    created: Actor.OwnedItem<I> | Array<Actor.OwnedItem<I>>,
+    { temporary }?: { temporary?: boolean }
+  ): ActiveEffect.Data[] | ActiveEffect.Data;
 
   /** @override */
   _onCreateEmbeddedEntity(
     embeddedName: string,
-    child: Actor.OwnedItem<I> | ActiveEffect,
+    child: Actor.OwnedItem<I> | ActiveEffect.Data,
     options: any,
     userId: string
   ): void;
 
   /** @override */
-  deleteEmbeddedEntity(embeddedName: string, data: string, options?: any): Promise<Actor.OwnedItem<I> | ActiveEffect>;
+  deleteEmbeddedEntity(
+    embeddedName: string,
+    data: string,
+    options?: any
+  ): Promise<Actor.OwnedItem<I> | ActiveEffect.Data>;
```

Comment:
See other comments.

Reply:
Improved it by overloading on `embeddedName`.